File: MAG/UI/mag_user/lib/models/rating.dart
import 'package:json_annotation/json_annotation.dart';
part 'rating.g.dart';

@JsonSerializable()
class Rating {
  int? id;
  int? userId;
  int? animeId;
  int? ratingValue;
  String? reviewText;
  DateTime? dateAdded;

  Rating(
    this.id,
    this.userId,
    this.animeId,
    this.ratingValue,
    this.reviewText,
    this.dateAdded,
  );

  factory Rating.fromJson(Map<String, dynamic> json) => _$RatingFromJson(json);

  Map<String, dynamic> toJson() => _$RatingToJson(this);
}

File: MAG/MAG/Database/Rating.cs
using System;
using System.Collections.Generic;

namespace MAG.Database;

public partial class Rating
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public int AnimeId { get; set; }

    public int? RatingValue { get; set; }

    public string ReviewText { get; set; } = null!;

    public DateTime DateAdded { get; set; }

    public virtual Anime Anime { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
